python/cuml/decomposition/pca.pyx

```diff
@@ -554,6 +555,7 @@ class PCA(Base):
 
         """
 
+        self._check_is_fitted('components_')
```

Comment:
I see that in sklearn, they do `sklearn.utils.validation.check_is_fitted` in a similar way. Is there an advantage to having this as a method on base instead of a utility? I believe we can always add the function later as a wrapper if necessary, so probably not a major issue either way.

Reply:
The advantage of a base method would be that there is no additionnal import needed, and it would be really easy to use. But then we would have two copy of this code, for the estimators with Dask.